tests/MongoDB.Driver.Tests/Specifications/unified-test-format/EntityMap.cs

```diff
@@ -0,0 +1,478 @@
+                                case "retryReads":
+                                    retryReads = option.Value.AsBoolean;
+                                    break;
+                                case "readConcernLevel":
+                                    var levelValue = option.Value.AsString;
+                                    var level = (ReadConcernLevel)Enum.Parse(typeof(ReadConcernLevel), levelValue, true);
+                                    readConcern = new ReadConcern(level);
+                                    break;
+                                case "w":
+                                    writeConcern = new WriteConcern(option.Value.AsInt32);
+                                    break;
+                                default:
+                                    throw new FormatException($"Unrecognized client uriOption name: '{option.Name}'");
+                            }
+                        }
+                        break;
+                    default:
+                        throw new FormatException($"Unrecognized client entity field: '{element.Name}'");
+                }
+            }
+
+            if (eventCapturer != null)
+            {
+                foreach (var eventTypeToCapture in eventTypesToCapture)
+                {
+                    switch (eventTypeToCapture)
+                    {
+                        case "commandStartedEvent":
+                            eventCapturer = eventCapturer.Capture<CommandStartedEvent>(x => !commandNamesToSkip.Contains(x.CommandName));
+                            break;
+                        case "commandSucceededEvent":
+                            eventCapturer = eventCapturer.Capture<CommandSucceededEvent>(x => !commandNamesToSkip.Contains(x.CommandName));
+                            break;
+                        case "commandFailedEvent":
+                            eventCapturer = eventCapturer.Capture<CommandFailedEvent>(x => !commandNamesToSkip.Contains(x.CommandName));
+                            break;
+                        default:
+                            throw new FormatException($"Invalid event name: {eventTypeToCapture}");
+                    }
+                }
+            }
+
+            var client = DriverTestConfiguration.CreateDisposableClient(
+                settings =>
+                {
+                    settings.RetryReads = retryReads;
+                    settings.RetryWrites = retryWrites;
+                    settings.ReadConcern = readConcern;
+                    settings.WriteConcern = writeConcern;
+                    settings.HeartbeatInterval = TimeSpan.FromMilliseconds(5); // the default value for spec tests
+                    if (eventCapturer != null)
+                    {
+                        settings.ClusterConfigurator = c => c.Subscribe(eventCapturer);
+                    }
+                },
+                useMultipleShardRouters);
```

Comment:
If you accept my suggestion to use named out variables instead of `Item1` and `Item2` (and you should) you will need a temporary local variable to work around CS1628:

    var localEventCapturer = eventCapturer; // copy value of eventCapturer ref variable to a local variable (to avoid error CS1628)
    client = DriverTestConfiguration.CreateDisposableClient(
        settings =>
        {
            settings.RetryReads = retryReads;
            settings.RetryWrites = retryWrites;
            settings.ReadConcern = readConcern;
            settings.WriteConcern = writeConcern;
            settings.HeartbeatInterval = TimeSpan.FromMilliseconds(5); // the default value for spec tests
            if (localEventCapturer != null)
            {
                settings.ClusterConfigurator = c => c.Subscribe(localEventCapturer);
            }
        },
        useMultipleShardRouters);